core/src/main/java/org/elasticsearch/search/suggest/SuggestBuilder.java

```diff
@@ -212,5 +227,68 @@ public T shardSize(Integer shardSize) {
             this.shardSize = shardSize;
             return (T)this;
         }
+
+        @Override
+        public String getWriteableName() {
+            return name;
+        }
+
+        @Override
+        public final T readFrom(StreamInput in) throws IOException {
+            T builder = doReadFrom(in);
+            builder.setName(in.readString());
+            builder.setSuggester(in.readString());
```

Comment:
I think we can get rid of this field in the abstract class, as far as I see it can be "term", "completion" or "phrase", those should be the NamedWritable NAME constants in the subclasses, so the superclass won't need to store it anymore.